R/app.R

```diff
@@ -0,0 +1,289 @@
+                    icon = shiny::icon("chart-line"), selected = NULL
+        ),
+        ## Crypto hehe
+        menuSubItem("Crypto Hehe",
+                    tabName = "crypto_wojak", href = NULL, newtab = TRUE,
+                    icon = shiny::icon("chart-line"), selected = NULL
+        )
+      )
+    )
+  ),
+  dashboardBody(
+    tabItems(
+      tabItem(
+        "stocks_tables",
+        fluidPage(
+          h1(icon("chart-line"), "Stock information by stock index"),
+          selectInput("stock_index",
+            label = "Stock Index:",
+            choices = tq_index_options()
+          ),
+          dataTableOutput("stocks_index_table")
+        ),
+        fluidPage(
+          h1(icon("chart-line"), "Stock information by stock exchange"),
+          selectInput("stock_exchange",
+            label = "Stock Exchange:",
+            choices = tq_exchange_options()
+          ),
+          dataTableOutput("stocks_exchange_table")
+        )
+      ),
+      tabItem(
+        "stocks_plot",
+        box(plotlyOutput("stock_price_plot"), width = 8),
+        box(selectInput("stock_name", "Stock name:",
+          selectize = FALSE,
+          choices = tq_exchange("NASDAQ")[1]
+        ), width = 4)
+      ),
+      tabItem(
+        "crypto_table",
+        fluidPage(
+          h1(icon("bitcoin"), "Crypto Currencies"),
+          dataTableOutput("cryptotable")
+        )
+      ),
+      tabItem(
+        "crypto_plots",
+        fluidPage(
+          h1(icon("chart-line"), "Top 10 crypto currencies"),
+          fluidRow(
+            column(12, 
+                   plotlyOutput("crypto1")),
+          ),
+          br(),
+          fluidRow(
+            column(4, 
+                   plotlyOutput("crypto2")),
+            column(4, 
+                   plotlyOutput("crypto3")),
+            column(4, 
+                   plotlyOutput("crypto4"))
+          ),
+          br(),
+          fluidRow(
+            column(4, 
+                   plotlyOutput("crypto5")),
+            column(4, 
+                   plotlyOutput("crypto6")),
+            column(4, 
+                   plotlyOutput("crypto7"))
+          ),
+          br(),
+          fluidRow(
+            column(4, 
+                   plotlyOutput("crypto8")),
+            column(4, 
+                   plotlyOutput("crypto9")),
+            column(4, 
+                   plotlyOutput("crypto10"))
+          )
+        )
+      ),
+      tabItem(
+        "crypto_wojak",
+        fluidPage(
+          h1(icon("bitcoin"), "Crypto Wojak"),
+          HTML('<iframe width="1120" height="630" src="https://www.youtube.com/embed/f4UoiE84mdA" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>')
+        )
+      )
+    )
+  )
+)
+
+# ------------------------------------------------------
+# SERVER
+#-------------------------------------------------------
+server <- function(input, output) {
```

Comment:
Maybe it would be a good idea to move `server` implementation to a different R file? What do you think?

Reply:
good idea, but we will have to duplicate all library( ) / #import? entries or very carefully cherrypick the neccessary ones.
Decision is up to you.